test/go-tests/sequencestate_test.go

```diff
@@ -0,0 +1,482 @@
+
+	cloudEvent = keptnv2.ToCloudEvent(*evaluationTriggeredEvent)
+
+	keptn, err = keptnv2.NewKeptn(&cloudEvent, keptncommon.KeptnOpts{EventSender: &APIEventSender{}})
+	require.Nil(t, err)
+
+	// send started event
+	_, err = keptn.SendTaskStartedEvent(nil, source)
+	require.Nil(t, err)
+
+	// send finished event with score
+	_, err = keptn.SendTaskFinishedEvent(&keptnv2.EvaluationFinishedEventData{
+		EventData: keptnv2.EventData{
+			Status: keptnv2.StatusSucceeded,
+			Result: keptnv2.ResultPass,
+		},
+		Evaluation: keptnv2.EvaluationDetails{
+			Score: 100.0,
+		},
+	}, source)
+	require.Nil(t, err)
+
+	// verify state
+	require.Eventually(t, func() bool {
+		states, resp, err = getState(projectName)
+		if err != nil {
+			return false
+		}
+		state := states.States[0]
+
+		if !isEqual(t, "state.State", "triggered", state.State) {
+			return false
+		}
+
+		if !isEqual(t, "len(state.Stages)", 2, len(state.Stages)) {
+			return false
+		}
+
+		devStage := state.Stages[0]
+
+		if !isEqual(t, "devStage.LatestEvaluation.Score", 100.0, devStage.LatestEvaluation.Score) {
+			return false
+		}
+
+		if !isEqual(t, "devStage.LatestEvent.Type", keptnv2.GetFinishedEventType("dev.delivery"), devStage.LatestEvent.Type) {
+			return false
+		}
+
+		stagingStage := state.Stages[1]
+
+		if !isEqual(t, "stagingStage.LatestEvent.Type", keptnv2.GetTriggeredEventType(keptnv2.DeploymentTaskName), stagingStage.LatestEvent.Type) {
+			return false
+		}
+
+		return true
+	}, 10*time.Second, 2*time.Second)
+
+	deploymentTriggeredEvent, err = getLatestEventOfType(*context.KeptnContext, projectName, "staging", keptnv2.GetTriggeredEventType(keptnv2.DeploymentTaskName))
+
+	require.Nil(t, err)
+	require.NotNil(t, deploymentTriggeredEvent)
+
+	cloudEvent = keptnv2.ToCloudEvent(*deploymentTriggeredEvent)
+
+	keptn, err = keptnv2.NewKeptn(&cloudEvent, keptncommon.KeptnOpts{EventSender: &APIEventSender{}})
+	require.Nil(t, err)
+
+	// send started event
+	_, err = keptn.SendTaskStartedEvent(nil, source)
+	require.Nil(t, err)
+
+	// send finished event with result=fail
+	_, err = keptn.SendTaskFinishedEvent(&keptnv2.EventData{
+		Status: keptnv2.StatusSucceeded,
+		Result: keptnv2.ResultFailed,
+	}, source)
+	require.Nil(t, err)
+
+	// verify state
+	require.Eventually(t, func() bool {
+		states, resp, err = getState(projectName)
+		if err != nil {
+			return false
+		}
+		state := states.States[0]
+
+		if !isEqual(t, "state.State", "finished", state.State) {
+			return false
+		}
+
+		if !isEqual(t, "len(state.Stages)", 2, len(state.Stages)) {
+			return false
+		}
+
+		stagingStage := state.Stages[1]
+
+		if !isEqual(t, "stagingStage.LatestEvent.Type", keptnv2.GetFinishedEventType("staging.delivery"), stagingStage.LatestEvent.Type) {
+			return false
+		}
+
+		return true
+	}, 10*time.Second, 2*time.Second)
+}
+
+func getLatestEventOfType(keptnContext, projectName, stage, eventType string) (*models.KeptnContextExtendedCE, error) {
```

Comment:
would it be possible to already now move and export these helper function to a separate file (e.g. `test-utils`) so that the actual integration test here only contains the test function and other functions which are absolutely specific for that test?

Reply:
If we decide to continue writing our integration tests in go in general, that would make perfect sense

Reply:
Agreed, this is a helpful function.